bot/exts/evergreen/wikipedia.py

```diff
@@ -42,43 +43,38 @@ class WikipediaSearch(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
-    async def wiki_request(self, channel: TextChannel, search: str) -> Optional[List[str]]:
+    async def wiki_request(self, channel: TextChannel, search: str) -> List[str]:
         """Search wikipedia search string and return formatted first 10 pages found."""
         params = WIKI_PARAMS | {"srlimit": 10, "srsearch": search}
         async with self.bot.http_session.get(url=SEARCH_API, params=params) as resp:
-            if resp.status == 200:
-                raw_data = await resp.json()
-                number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
-
-                if number_of_results:
-                    results = raw_data["query"]["search"]
-                    lines = []
-
-                    for article in results:
-                        line = WIKI_SEARCH_RESULT.format(
-                            name=article["title"],
-                            description=unescape(
-                                re.sub(
-                                    WIKI_SNIPPET_REGEX, "", article["snippet"]
-                                )
-                            ),
-                            url=f"https://en.wikipedia.org/?curid={article['pageid']}"
-                        )
-                        lines.append(line)
-
-                    return lines
-
-                else:
-                    await channel.send(
-                        "Sorry, we could not find a wikipedia article using that search term."
-                    )
-                    return
-            else:
+            if resp.status != 200:
                 log.info(f"Unexpected response `{resp.status}` while searching wikipedia for `{search}`")
-                await channel.send(
-                    "Whoops, the Wikipedia API is having some issues right now. Try again later."
-                )
-                return
+                raise APIError("Wikipedia API", resp.status)
+
+            raw_data = await resp.json()
+
+            if not raw_data.get("query"):
+                if error := raw_data.get("errors"):
+                    log.error(f"There was an error while communicating with the Wikipedia API: {error}")
+                raise APIError("Wikipedia API", resp.status, error)
+
+            lines = []
+            if raw_data["query"]["searchinfo"]["totalhits"]:
+                results = raw_data["query"]["search"]
+
+                for article in results:
```

Comment:
```suggestion
                for article in raw_data["query"]["search"]:
```

Reply:
I pushed the changes, although the only thing I changed in the code was a few checks, that was the base code, it made it a bit more consistent, but I understand that those variables were unnecessary.